python/vision_explanation_methods/error_labeling/error_labeling.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the Error Labeling Manager class."""
+
+from enum import Enum
+
+import numpy as np
+import torchvision
+from torch import Tensor
+
+LABELS = 'labels'
+
+
+class ErrorLabelType(Enum):
+    """
+    Enum providing types of error labels.
+
+    If none, then the detection is not an error. It is a
+    correct prediction.
+    """
+
+    # the gt doesn't have a corresponding detection
+    MISSING = "missing"
+
+    # the model predicted detections, but there was nothing there
+    # this prediction must have a 0 iou score with all gt detections
+    BACKGROUND = "background"
+
+    # the predicted class is correct, bounding box is not
+    LOCALIZATION = "localization"
+
+    # the predicted class is incorrect, the bounding box is correct
+    CLASS_NAME = "class_name"
+
+    # both the predicted class and bounding box are incorrect
+    CLASS_LOCALIZATION = "class_localization"
+
+    # the predicted class is correct, the bounding box is correct, but
+    # the iou score is lower than another detection
+    DUPLICATE_DETECTION = "duplicate_detection"
+
+    MATCH = "match"
+
+
+class ErrorLabeling():
+    """Defines a wrapper class of Error Labeling for vision scenario.
+
+    Only supported for object detection at this point.
+    """
+
+    def __init__(self,
+                 task_type: str,
+                 pred_y: str,
+                 true_y: str,
+                 iou_threshold: float = 0.5):
+        """Create an ErrorLabeling object.
+
+        :param model: The model to explain.
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param evaluation_examples: A matrix of feature vector
+            examples (# examples x # features) on which to explain the
+            model's output, with an additional label column.
+        :type evaluation_examples: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param task_type: The task to run.
```

Comment:
I'm not sure if we need this task_type.  Maybe keep it as is for now?  I just don't know if we will really need this for other tasks.  I guess the other ones we have are much simpler right now, perhaps with image segmentation they will be more complicated again and will require this.